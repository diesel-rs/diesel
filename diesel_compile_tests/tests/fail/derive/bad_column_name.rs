#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
        #[sql_name = "spa ce"]
        space -> Text,
    }
}

#[derive(Queryable)]
#[diesel(table_name = users)]
struct User1 {
    #[diesel(column_name)]
    //~^ ERROR: unexpected end of input, expected `=`
    name: String,
}

#[derive(Queryable)]
#[diesel(table_name = users)]
struct User2 {
    #[diesel(column_name(another))]
    //~^ ERROR: expected `=`
    name: String,
}

#[derive(Queryable)]
#[diesel(table_name = users)]
struct User3 {
    #[diesel(column_name = true)]
    //~^ ERROR: expected string literal
    name: String,
}

#[derive(Insertable)]
#[diesel(table_name = users)]
struct User4 {
    #[diesel(column_name = "spa ce")]
    //~^ ERROR:  expected valid identifier, found `spa ce`. Diesel does not support column names with whitespaces yet
    space: String,
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
struct User5 {
    #[diesel(column_name = "spa ce")]
    //~^ ERROR:  expected valid identifier, found `spa ce`. Diesel does not support column names with whitespaces yet
    space: String,
}

fn main() {}
