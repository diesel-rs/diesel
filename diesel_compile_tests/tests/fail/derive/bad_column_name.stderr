error: unexpected end of input, expected `=`
       help: The correct format looks like `#[diesel(column_name = foo)]`
  --> tests/fail/derive/bad_column_name.rs:16:25
   |
16 |     #[diesel(column_name)]
   |                         ^

error: expected `=`
  --> tests/fail/derive/bad_column_name.rs:23:25
   |
23 |     #[diesel(column_name(another))]
   |                         ^

error: expected string literal
  --> tests/fail/derive/bad_column_name.rs:30:28
   |
30 |     #[diesel(column_name = true)]
   |                            ^^^^

error: Expected valid identifier, found `type`. Diesel automatically renames invalid identifiers, perhaps you meant to write `type_`?
  --> tests/fail/derive/bad_column_name.rs:38:28
   |
38 |     #[diesel(column_name = "type")]
   |                            ^^^^^^

error: Expected valid identifier, found `type`. Diesel automatically renames invalid identifiers, perhaps you meant to write `type_`?
  --> tests/fail/derive/bad_column_name.rs:46:28
   |
46 |     #[diesel(column_name = "type")]
   |                            ^^^^^^
